django/db/models/base.py

```diff
@@ -1796,6 +1798,22 @@ def _check_field_name_clashes(cls):
             used_fields[f.name] = f
             used_fields[f.attname] = f
 
+        # Additional check for OneToOnefield
+        # Not listed in cls._meta.local_fields
+        for local_f in cls._meta.parents.values():
+            if not local_f:
+                continue
+            clash = parent_fields.get(local_f.name) or None
+            if clash:
+                errors.append(
+                    checks.Error(
+                        "The field '%s' clashes with the field '%s' "
+                        "from model '%s'."
+                        % (local_f.name, clash.name, clash.model._meta),
+                        obj=local_f,
```

Comment:
I would expect to `ItalianRestaurant._meta.get_field("place_ptr")` here but it's `<django.db.models.fields.related.OneToOneField: place_ptr>` which is not very helpful for users as it's not attached to the model.

IMO, it would be better to return `ItalianRestaurant._meta.get_field("place_ptr")` even if it's the same as `Restaurant._meta.get_field("place_ptr")`.

Reply:
Message is fine, but `obj` is not. As far as I'm aware, it should be `ItalianRestaurantManyParents._meta.get_field("place_ptr")` or `Restaurant._meta.get_field("place_ptr")` :thinking:

Reply:
Agreed.